eessi_container.sh

```diff
@@ -756,14 +756,40 @@ for cvmfs_repo in "${REPOSITORIES[@]}"
 do
     unset cfg_repo_id
     [[ ${VERBOSE} -eq 1 ]] && echo "add fusemount options for CVMFS repo '${cvmfs_repo}'"
-    # split into name and access mode if ',access=' in $cvmfs_repo
-    if [[ ${cvmfs_repo} == *",access="* ]] ; then
-        cvmfs_repo_name=${cvmfs_repo/,access=*/} # remove access mode specification
-        cvmfs_repo_access=${cvmfs_repo/*,access=/} # remove repo name part
-    else
-        cvmfs_repo_name="${cvmfs_repo}"
-        cvmfs_repo_access="${ACCESS}" # use globally defined access mode
+    # split into name, access mode, and mount mode
+    readarray -td, cvmfs_repo_args <<<"$cvmfs_repo"
+    cvmfs_repo_name=${cvmfs_repo_args[0]}
+    cvmfs_repo_access="${ACCESS}" # initialize to the default access mode
+    for arg in ${cvmfs_repo_args[@]:1}; do
+        if [[ $arg == "access="* ]]; then
+            cvmfs_repo_access=${arg/access=}
+        fi
+        if [[ $arg == "mount="* ]]; then
+            cvmfs_repo_mount=${arg/mount=}
+            # check if the specified mount mode is a valid one
+            if [[ ${cvmfs_repo_mount} != "bind" ]] && [[ ${cvmfs_repo_mount} != "fuse" ]]; then
+                echo -e "ERROR: mount mode '${cvmfs_repo_mount}' for CVMFS repository\n  '${cvmfs_repo_name}' is not known"
+                exit ${REPOSITORY_ERROR_EXITCODE}
+            fi
+        fi
+    done
+    # if a mount mode was not specified, we use a bind mount if the repository is available on the host,
+    # and otherwise we use a fuse mount
+    if [[ -z ${cvmfs_repo_mount} ]]; then
+        cvmfs_repo_mount="fuse"
+        if [[ -x $(command -v cvmfs_config) ]] && cvmfs_config probe ${cvmfs_repo_name} >& /dev/null; then
+            cvmfs_repo_mount="bind"
```

Comment:
@bedroge Hmm, not sure I like this, since it changes the default behavior.

I think it's better to only to bind mount if explicitly requested, to retain same (default) behavior across different systems.

Auto-enabling the bind mount when the CernVM-FS repo is mounted leaves us vulnerable in case of trouble with that mount (like an outdated sync, CernVM-FS cache trouble, etc.)

Reply:
Changed in https://github.com/EESSI/software-layer-scripts/pull/57.